personalities/ct-static-api/instance.go

```diff
@@ -138,30 +117,28 @@ func setUpLogInfo(ctx context.Context, opts InstanceOptions) (*logInfo, error) {
 	}
 
 	var signer crypto.Signer
-	if !cfg.IsMirror {
-		var err error
-		if signer, err = keys.NewSigner(ctx, vCfg.PrivKey); err != nil {
-			return nil, fmt.Errorf("failed to load private key: %v", err)
-		}
+	var err error
+	if signer, err = keys.NewSigner(ctx, vCfg.PrivKey); err != nil {
+		return nil, fmt.Errorf("failed to load private key: %v", err)
+	}
 
-		// If a public key has been configured for a log, check that it is consistent with the private key.
-		if vCfg.PubKey != nil {
-			switch pub := vCfg.PubKey.(type) {
-			case *ecdsa.PublicKey:
-				if !pub.Equal(signer.Public()) {
-					return nil, errors.New("public key is not consistent with private key")
-				}
-			case ed25519.PublicKey:
-				if !pub.Equal(signer.Public()) {
-					return nil, errors.New("public key is not consistent with private key")
-				}
-			case *rsa.PublicKey:
-				if !pub.Equal(signer.Public()) {
-					return nil, errors.New("public key is not consistent with private key")
-				}
-			default:
-				return nil, errors.New("failed to verify consistency of public key with private key")
+	// If a public key has been configured for a log, check that it is consistent with the private key.
+	if vCfg.PubKey != nil {
+		switch pub := vCfg.PubKey.(type) {
+		case *ecdsa.PublicKey:
+			if !pub.Equal(signer.Public()) {
+				return nil, errors.New("public key is not consistent with private key")
+			}
+		case ed25519.PublicKey:
+			if !pub.Equal(signer.Public()) {
+				return nil, errors.New("public key is not consistent with private key")
 			}
+		case *rsa.PublicKey:
```

Comment:
I think you can drop support for `rsa` as there's no mechanism for using it to sign notes. Similarly, I suspect CT Static API doesn't provide for `Ed25519` signatures on SCTs or Checkpoints for backwards compatibility reasons, so I think you're probably down to just `ECDSA` here.

Reply:
Yeah, I was wondering why it's here in the first place. It got added in https://github.com/google/certificate-transparency-go/commit/0f11cd0b1754a2129c046894f6e436ab9467ad59. I wonder whether we can simply remove all mentions to public keys. It says somewhere that it's used for debugging / testing, but I'm willing to bet it's not actually used.

Reply:
If it stays here, this logic should really go in config.go though :)